[shopsys] BreadcrumbGenerators are now automatically registered in BreadcrumbResolver (#1141)
shopsys/shopsys: docs/introduction/friendly-url-breadcrumb-navigation.md

        - if the name of the entity will be changed, the old URLs will now redirect to your new URLs
- run `php phing friendly-urls-generate` to generate new URLs
- visit some of provided URLs and check if everything works fine

# Breadcrumb navigation

> All frontend routes include breadcrumb navigation in top of content page to ease navigation for your customers across your e-commerce platform.

Reviewer: ```suggestion
All frontend routes include breadcrumb navigation in the top of the page to ease the navigation for your customers across your e-commerce site.
```

> You might want to implement new `BreadcrumbGenerator` for your new frontend routes to tell application what navigation you want to display.

Reviewer: ```suggestion
When adding a new page on the frontend, you need to implement new `BreadcrumbGenerator` for the new routes to tell the application how the navigation should be displayed.
```

## How to create new `BreadcrumbGenerator`

- create new class with name ending with `BreadcrumbGenerator`
- this class has to implement `BreadcrumbGeneratorInterface`
- this interface requires you to implement two methods *(see `ArticleBreadcrumbGenerator` class as an example of the implementation)*:
    - `getBreadcrumbItems` method that generates `BreadcrumbItems`

>         - these include displayed name, route and route parameters

Reviewer: ```suggestion
        - these include displayed name, and may include route and route parameters if you want to make a link from the item
```

>     - `getRouteNames` method where you have to provide names of routes where you want this navigation to appear

Reviewer: ```suggestion
    - `getRouteNames` method where you have to provide names of the routes that you want to include in the navigation
```

>     - `getRouteNames` method where you have to provide names of the routes that you want to include in the navigation

Reviewer: ```suggestion
    - `getRouteNames` method where you have to provide names of the routes for which you want to use the your breadcrumb generator
```
